### ISSUE TITLE
Parsing source code

### ISSUE THREAD

**Issue by rubdos** [myint/language-check#29]
Cfr. vim-syntastic/syntastic/issues/1918.

When using syntastic with language-check on source files, program code is obviously parsed as language, and obviously has a lot of grammatical mistakes.

I know this is a big one, but would it be interesting to add some experimental support to language-check to parse source files? This would be especially useful for LaTeX, as that contains a lot of text. I also think it is useful to parse eg. C, C++ and Java code, and check comments for grammatical and spelling mistakes.

What do you think?

**Comment by myint** [comment]
I've used [`detex`](https://github.com/pkubowicz/opendetex) for this purpose before.

```
detex document.tex | \
    language-check \
    --language=en-US \
    --disable=COMMA_PARENTHESIS_WHITESPACE \
    --disable=HE_VERB_AGR \
    --disable=MORFOLOGIK_RULE_EN_US \
    --disable=THREE_NN \
    --disable=TO_NON_BASE \
    --disable=UPPERCASE_SENTENCE_START \
    --disable=WHITESPACE_RULE \
    -
```

This was just for a short resume. I use `detex` on larger documents to do spelling checking, but never with `language-check`. My guess is that you may get additional false positives when checking papers with figures and more complex things.

Thanks for pointing out the Syntastic issue. `proselint`, which `lcd047` mentioned, looks interesting.

**Comment by jxmorris12** [comment]
Hi! Since this project has been abandoned, I started a new fork over at https://github.com/jxmorris12/language_tool_python. 

My version supports new versions of Java and LanguageTool. `language-check` is stuck on Java 8 and LanguageTool 3.2; latest versions are Java 14 and LanguageTool 4.9!

I'm happy to help you with your issue if you raise it over at my repository! Thanks!
